src/chunk/chunk_unknown.rs

```diff
@@ -0,0 +1,50 @@
+use std::any::Any;
+use std::fmt::{Debug, Display, Formatter};
+use bytes::{Bytes, BytesMut};
+use crate::chunk::Chunk;
+use crate::chunk::chunk_header::{CHUNK_HEADER_SIZE, ChunkHeader};
+
+#[derive(Clone, Debug)]
+pub struct ChunkUnknown {
+    hdr: ChunkHeader,
+    value: Bytes,
+}
+
+impl Display for ChunkUnknown {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "ChunkUnknown( {} {:?} )", self.header(), self.value)
+    }
+}
+
+impl Chunk for ChunkUnknown {
+    fn header(&self) -> ChunkHeader {
+        self.hdr.clone()
+    }
+
+    fn as_any(&self) -> &(dyn Any + Send + Sync) {
+        self
+    }
+
+    fn check(&self) -> crate::error::Result<()> {
+        Ok(())
+    }
+
+    fn value_length(&self) -> usize {
+        self.value.len()
+    }
+
+    fn marshal_to(&self, buf: &mut BytesMut) -> crate::error::Result<usize> {
+        self.header().marshal_to(buf)?;
+        buf.extend(&self.value);
+        Ok(buf.len())
+    }
+
+    fn unmarshal(raw: &Bytes) -> crate::error::Result<Self> where Self: Sized {
+        let header = ChunkHeader::unmarshal(raw)?;
+        let len = header.value_length as usize;
+        Ok(Self {
+            hdr: header,
+            value: raw.slice(CHUNK_HEADER_SIZE..).slice(..len),
```

Comment:
```suggestion
            value: raw.slice(CHUNK_HEADER_SIZE..len),
```

?

Reply:
Your change would make the resulting slice just `len - CHUNK_HEADER_SIZE` long.

The correct change would be `value: raw.slice(CHUNK_HEADER_SIZE .. CHUNK_HEADER_SIZE + len),`

I did this out of the habit of doing this with normal slices, where it compiles to the same code. In this case it might actually make difference though. I'll change it.

```rust
let slice = &[1,2,3,4,5];
assert_eq!(
    &slice[1..1+3],    // normal slicing, length is somewhat hidden inside some noise
    &slice[1..][..3]    // double slicing, length is easy to see
);
```